### PR TITLE
Modernize switch statements

### PR DESCRIPTION
**Pull request by richlander**
Switch statements were converted to more terse switch expressions.

Most (or all) of the remaining switch statements are not candidates for switch expressions.

I also converted a number of 1-line methods to method body expressions as well.

### CODE REVIEW

**Review by Ellerbach — APPROVED** [review]
This new switch notation makes the code very easy to read and concise. The more I look at it, the more I really like it!

File: src/devices/Mcp25xxx/Register/MessageTransmit/TxBxDn.cs [hunk 1]
```diff
@@ -60,32 +60,12 @@ public TxBxDn(byte txBufferNumber, int index, byte transmitBufferDataFieldBytes)
...
-                case Address.TxB1D4:
-                case Address.TxB1D5:
-                case Address.TxB1D6:
-                case Address.TxB1D7:
-                    return 1;
-                default:
-                    throw new ArgumentException($"Invalid address value {address}.", nameof(address));
-            }
-        }
+            Address.TxB0D0 or Address.TxB0D1 or Address.TxB0D2 or Address.TxB0D3 or Address.TxB0D4 or Address.TxB0D5 or Address.TxB0D6 or Address.TxB0D7 => 0,
```
Comment (krwq) [inline src/devices/Mcp25xxx/Register/MessageTransmit/TxBxDn.cs:65]: this specific case I find old syntax more readable

Comment (krwq) [reply src/devices/Mcp25xxx/Register/MessageTransmit/TxBxDn.cs:65]: is it possible to specify ranges? (>= && <=)

**Review by krwq — APPROVED** [review]

File: src/devices/Mcp25xxx/Register/MessageTransmit/TxBxDn.cs (hunk 1 above)
Comment (richlander) [reply src/devices/Mcp25xxx/Register/MessageTransmit/TxBxDn.cs]: How does this look? https://github.com/dotnet/iot/pull/1303/commits/3516049c0322dec4a5bb9038fecf88cfca842bd9

Comment (krwq) [reply src/devices/Mcp25xxx/Register/MessageTransmit/TxBxDn.cs]: nice :)

**Review by krwq — APPROVED** [review]

**Pull request merged**